### PR TITLE
fix(CBH): fix cbh instance eip

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  fix cbh instance eip
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix cbh instance eip
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic
--- PASS: TestAccCBHInstance_basic (1075.79s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1075.831s
```

### CODE REVIEW

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 1]
```diff
@@ -673,6 +672,19 @@ func unbindEip(d *schema.ResourceData, client *golangsdk.ServiceClient, publicIp
...
+		if apiErr, ok := err.(golangsdk.ErrDefault400); ok {
+			var respBody interface{}
+			if jsonErr := json.Unmarshal(apiErr.Body, &respBody); jsonErr != nil {
+				return jsonErr
+			}
+			errCode := utils.PathSearch("error_code", respBody, "")
+			// these two error code indicate the eip has been unbound or deleted
+			if errCode == "CBH.10020010" || errCode == "CBH.10020009" {
+				log.Printf("[WARN] Failed to unbind EIP (ID: %s， IP: %s) from CBH instance: %s", publicIpId,
+					publicIp, d.Id())
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]:
```suggestion
					publicIp, apiErr)
```

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go [hunk 2]
```diff
@@ -763,7 +775,9 @@ func resourceCBHInstanceRead(_ context.Context, d *schema.ResourceData, meta int
 		if publicIpId != nil && strings.TrimSpace(publicIpId.(string)) != "" {
 			publicIp, err = getPublicAddressById(d, cfg, strings.TrimSpace(publicIpId.(string)))
 			if err != nil {
-				return diag.FromErr(err)
+				if _, ok := err.(golangsdk.ErrDefault404); !ok {
+					return diag.FromErr(err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:779]: should we clear **publicIpId** when the error code is 404?

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go]: fixed

File: huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/cbh/resource_huaweicloud_cbh_instance.go:779]: done

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/cbh/' TESTARGS='-run TestAccCBHInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbh/ -v -run TestAccCBHInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccCBHInstance_basic
=== PAUSE TestAccCBHInstance_basic
=== CONT  TestAccCBHInstance_basic

--- PASS: TestAccCBHInstance_basic (1154.06s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbh       1154.099s

**Pull request merged**